internal/controllers/gateway/route_utils.go

```diff
@@ -500,17 +506,6 @@ func existsMatchingReadyListenerInStatus[T gatewayapi.RouteT](route T, listener
 		return errUnsupportedRouteKind // Listener(s) found but none with matching supported kinds.
 	}
 
-	// ... and verify if it's programmed.
-	lReadyCond, ok := lo.Find(listenerStatus.Conditions, func(c metav1.Condition) bool {
-		return c.Type == string(gatewayapi.ListenerConditionProgrammed)
-	})
-	if !ok {
-		return errListenerNotProgrammed
-	}
-	if lReadyCond.Status != "True" {
-		return errListenerNotReadyYet // Listener is not ready yet.
-	}
```

Comment:
Shouldn't we verify if the `Listener` is `Accepted`?

Reply:
Reading https://github.com/kubernetes-sigs/gateway-api/pull/2535 it appears it doesn't have to be `Accepted` so sounds right to skip it.

Reply:
According to the latest version of gateway API, the `HTTPRoute` should have `Accepted` set to `false`.

Reply:
By digging the history of issues/PRs in gateway-api repo, https://github.com/kubernetes-sigs/gateway-api/pull/2396 clarifies that the `attachedRoutes` should be added when the listener allows the route by `AllowedRoutes` and gateway is present in route's parent refs, regardless of status of listener or route:
```
	// Successful attachment of a Route to a Listener is based solely on the
	// combination of the AllowedRoutes field on the corresponding Listener
	// and the Route's ParentRefs field
```